Tests/PenguinStructuresTests/EitherTests.swift

```diff
@@ -0,0 +1,39 @@
+//******************************************************************************
+// Copyright 2020 Penguin Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+import PenguinStructures
+
+
+class EitherTests: XCTestCase {
+  
+  func testEquality() {
+    XCTAssertEqual(Either<Int, String>.a(1), .a(1))
+    XCTAssertNotEqual(Either.a(1), .b(1))
+    XCTAssertNotEqual(Either.a(1), .b("1"))
+    XCTAssertEqual(Either<Int, String>.b("0"), .b("0"))
+  }
+
+  func testComparable() {
+    typealias E = Either<Int, Int>
+    XCTAssert(E.a(1) < .b(0))
+    XCTAssert(E.b(0) > .a(100000))
```

Comment:
There are required semantic relationships between the Comparable and Equatable operations.  Also see note on `testEquality`.